components/metadata-service/internal/metadata/secrets/service.go

```diff
@@ -0,0 +1,127 @@
+package secrets
+
+import (
+	"github.com/kyma-project/kyma/components/metadata-service/internal/apperrors"
+)
+
+const (
+	OauthClientIDKey     = "clientId"
+	OauthClientSecretKey = "clientSecret"
+
+	BasicAuthUsernameKey = "username"
+	BasicAuthPasswordKey = "password"
+)
+
+type Service interface {
+	CreateOauthSecret(remoteEnvironment, name, clientID, clientSecret, serviceID string) apperrors.AppError
```

Comment:
This design could be simplified. Currently serviceapiservice is responsible for detecting what is the type of credentials and creating secret name. It could be a responsibility of secrets.service.
Proposed solution:
1) creating a new package for data model 
2) moving all definitions from serviceapi/model.go to the new package
3) importing the model package is secrets/service.go
4) Change secrets.Service interface to have three methods (Create, Update, Delete) . Create and update would take two arguments: remote environment name and credentials object (so that it could figure out the type of credentials and secret name)

With such a design serviceapiservice delegates all the details of secretes handling to secrets.service. It gets shorter and also the tests gets easier.